File: server/moneyflow/src/main/java/com/moneyflow/moneyflow/enums/TipoRendimento.java
package com.moneyflow.moneyflow.enums;

import lombok.RequiredArgsConstructor;
import org.apache.commons.lang3.EnumUtils;

import java.util.stream.Collectors;

@RequiredArgsConstructor
public enum TipoRendimento {

	RECEITA(1L, "Receita"),
	DESPESA(2L, "Despesa");

	private final Long codigo;
	private final String descricao;

	public Long getCodigo () {
		return codigo;
	}

	public String getDescricao () {
		return descricao;
	}

	public static TipoRendimento getByValue (Long value) {
		return EnumUtils.getEnumList(TipoRendimento.class).stream()
				.collect(Collectors.toMap(TipoRendimento::getCodigo, tipoRendimento -> tipoRendimento))
				.get(value);
	}

}

File: web/src/app/base/tipo-rendimento.ts
export enum TipoRendimento {
  RECEITA = "RECEITA",
  DESPESA = "DESPESA"
}

export class TipoRendimentoUtil {

  public static getFieldByTipoRendimento(tipoRendimento: string): string {
    return tipoRendimento ? tipoRendimento.charAt(0).concat(tipoRendimento.toLowerCase().substring(1)) : '';
  }

}
